search/engine.go

```diff
@@ -528,18 +547,39 @@ func (e *ESEngine) DoSearch(ctx context.Context, query Query, sortBy string, fro
 		mssHighlights := e.esc.MultiSearch()
 		highlightRequestAdded := false
 
+		highlightsLangs := query.LanguageOrder
+		if !shouldMergeResults {
+			highlightsLangs = []string{currentLang}
+		}
+
 		for _, h := range ret.Hits.Hits {
 
+			if h.Type == consts.SEARCH_RESULT_TWEETS_MANY && h.InnerHits != nil {
+				if tweetHits, ok := h.InnerHits[consts.SEARCH_RESULT_TWEETS_MANY]; ok {
+					for _, th := range tweetHits.Hits.Hits {
+						mssHighlights.Add(NewResultsSearchRequest(
```

Comment:
Do we run 20 highlight searches here?
Should we run 1 highlight search for all tweets?

How do we do this with non-tweet results?

Reply:
I have a concert about too many requests. Even if they take less time as later this might overload the server too much. 

What do you think?

Maybe reduce to 10 tweets?